protocol/messenger_handler.go

```diff
@@ -1594,21 +1595,29 @@ func (m *Messenger) HandleCommunityRequestToJoinResponse(state *ReceivedMessageS
 		return ErrInvalidCommunityID
 	}
 
-	updatedRequest, err := m.communitiesManager.HandleCommunityRequestToJoinResponse(signer, requestToJoinResponseProto)
+	myCancelledRequestToJoin, err := m.MyCanceledRequestToJoinForCommunityID(requestToJoinResponseProto.CommunityId)
+
 	if err != nil {
 		return err
 	}
 
-	if updatedRequest != nil {
-		state.Response.RequestsToJoinCommunity = append(state.Response.RequestsToJoinCommunity, updatedRequest)
+	// Updating the request if request it was cancelled breaks the test
+	if myCancelledRequestToJoin.State == 0 {
```

Comment:
1) Please use `nil` assertion as suggested before. Comparing State to 0 is not readable at all.
2) Please use early return instead of populating each block with this condition.

```suggestion
	// Don't process response if we cancelled request to join.
	if myCancelledRequestToJoin != nil {
            return nil
    }
```